README.md

```diff
@@ -0,0 +1,536 @@
+# MQTT for Elixir
+
+## Introduction
+
+Recently I have read an excellent book [Build a Weather Station with Elixir and Nerves](https://pragprog.com/titles/passweather/build-a-weather-station-with-elixir-and-nerves/).
+It introduces Elixir as a tool for building embedded applications.
+
+With [Nerves](https://www.nerves-project.org/), we can
+run Elixir code on network-enabled devices interacting with
+some controlling software and each other.
+
+The book mentioned above mainly focuses on the Nerves part and
+uses HTTP protocol for network interactions. Although this is
+a reasonable choice in many situations, I want to introduce another
+widely used option for production IoT setups: [MQTT](https://mqtt.org/).
+
+## MQTT Protocol
+
+[MQTT](https://mqtt.org/) is a messaging protocol designed specifically for (IoT) device
+communications. It is used in many areas, such as banking, oil & gas industry, manufacturing, etc.
+
+MQTT protocol has many advantages, but here I want to mention some of them:
+
+* It is a lightweight binary protocol generally running over TCP/IP.
+* It is designed for unreliable networking, making it a good choice for outdoor installations.
+* It follows pub/sub model simplifying client logic.
+
+We will demonstrate some of MQTT's advantages in our setup.
+
+## MQTT Brokers
+
+An essential thing about MQTT is that it simplifies the client's logic, which is critical for embedded devices.
+This is achieved with the pubsub model: in MQTT, there is no concept of a "server".
+Instead, all participating entities are clients connecting to a so-called _broker_.
+Clients _subscribe_ to _topics_, _publish_ messages to them, and the broker does the routing (and many other things).
+
+A good production-ready broker, like [EMQX](https://github.com/emqx/emqx) generally provides
+not only MQTT routing capabilities but many other intellectual features, like
+* other kinds of connection methods, like WebSockets;
+* different models of authentication and authorization;
+* streaming data to databases;
+* custom routing rules based on message introspection;
+* and so on.
```

Comment:
Maybe it's interesting to mention persistent sessions and retained messages?  It's something unusual in the Req/Rep or HTTP world.

Reply:
I thought that for a ordinary reader this features a too specific; we demonstrate retaining further though.